src/Api/Endpoints/ProductEndpoints.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MediatR;
+using Mapster;
+using Application.Common.Exceptions;
+using Application.Products.Queries.GetAllProducts;
+using Application.Products.Commands.AddProduct;
+using Application.Products.Commands.UpdateProduct;
+using Application.Products.Commands.DeleteProduct;
+using Application.Products.Queries.GetProduct;
+using Api.Contracts.Responses;
+using Api.Contracts.Requests;
+
```

Comment:
I strongly believe that our namespaces are already too granular and that "namespaces = folder paths" convention is a code smell. Look at that `using` list - and we've barely started!

Reply:
Two thoughts:

1. As this is a template, when using Visual Studio or Rider/ReSharper, once the root namespace is chosen, if automatically renamed, all the namespaces will be updated to its consistent folder structure. So it's better to have those using already pointing to that structure.
2.  Also, to avoiding default warnings (yeah, could be disabled) about not matching namespaces, I'll also stick to default folder structure.

All that said, I see some places where it could be interesting to skip that. And of course, I'm also onto that if enough reasons and value :)

Reply:
The compromise convention I use is that the namespace _must_ reflect the folder structure, but you are free to skip any amount of sub-namespaces from the end, e.g.
Solution `Collaboration.OpenTemplate`, project `Application`, folder `CQRS\Requests\AddProduct`, file `AddproductCommand.cs` can be in either of these namespaces:
- `Collaboration.OpenTemplate.Application.CQRS.Requests.AddProduct`
- `Collaboration.OpenTemplate.Application.CQRS.Requests`
- `Collaboration.OpenTemplate.Application.CQRS`
- `Collaboration.OpenTemplate.Application`
- `Collaboration.OpenTemplate`
And then I just tend towards the shorter ones (results in less sub-namespaces).

Usually I start with one global namespace, then if it gets too crowded, I create a sub-namespace for the biggest group of files that can benefit from a sub-namespace. Repeat when needed.

Reply:
Anyways, we can change it anytime, and I don't want it to be blocking this PR, so I'll resolve this discussion.